--- ../qemu/s2e/S2EExecutor.cpp.orig	2016-12-07 14:59:42.820079828 +0000
+++ ../qemu/s2e/S2EExecutor.cpp	2016-12-07 15:00:08.210079914 +0000
@@ -646,11 +646,11 @@ S2EExecutor::S2EExecutor(S2E* s2e, TCGLL
     //__DEFINE_EXT_FUNCTION(raise_exception)
     //__DEFINE_EXT_FUNCTION(raise_exception_err)
 
-#ifdef _WIN32
+// #ifdef _WIN32
     __DEFINE_EXT_VARIABLE(g_s2e_concretize_io_addresses)
     __DEFINE_EXT_VARIABLE(g_s2e_concretize_io_writes)
     __DEFINE_EXT_VARIABLE(g_s2e_fork_on_symbolic_address)
-#endif
+// #endif
 
     __DEFINE_EXT_VARIABLE(g_s2e_enable_mmio_checks)
 
